File: src/main/java/com/telegram/codex/interfaces/cli/CliTaskRunner.java
package com.telegram.codex.interfaces.cli;

import com.telegram.codex.integration.telegram.application.port.out.TelegramGateway;
import com.telegram.codex.integration.telegram.domain.TelegramBotCommand;
import com.telegram.codex.shared.config.AppProperties;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.CommandLineRunner;
import org.springframework.context.ConfigurableApplicationContext;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class CliTaskRunner implements CommandLineRunner {

    private final ConfigurableApplicationContext applicationContext;
    private final AppProperties properties;
    private final TelegramGateway telegramClient;
    private final String task;

    public CliTaskRunner(
        ConfigurableApplicationContext applicationContext,
        AppProperties properties,
        TelegramGateway telegramClient,
        @Value("${app.task:}") String task
    ) {
        this.applicationContext = applicationContext;
        this.properties = properties;
        this.telegramClient = telegramClient;
        this.task = task;
    }

    @Override
    public void run(String... args) {
        if (task == null || task.isBlank()) {
            return;
        }
        if ("telegram:set-webhook".equals(task)) {
            telegramClient.setWebhook(properties.getBaseUrl() + "/telegram/webhook", properties.getTelegramWebhookSecret());
        } else if ("telegram:update-commands".equals(task)) {
            telegramClient.setMyCommands(List.of(
                new TelegramBotCommand("status", "Bot 狀態"),
                new TelegramBotCommand("session", "目前 session 狀態"),
                new TelegramBotCommand("memory", "長期記憶狀態"),
                new TelegramBotCommand("forget", "清除長期記憶"),
                new TelegramBotCommand("compact", "壓縮目前對話 context"),
                new TelegramBotCommand("new", "新 session"),
                new TelegramBotCommand("help", "使用說明")
            ));
        } else {
            throw new IllegalArgumentException("Unknown task: " + task);
        }
        System.exit(org.springframework.boot.SpringApplication.exit(applicationContext, () -> 0));
    }
}

File: src/main/kotlin/com/telegram/codex/interfaces/cli/CliTaskRunner.kt
package com.telegram.codex.interfaces.cli

import com.telegram.codex.integration.telegram.application.port.out.TelegramGateway
import com.telegram.codex.integration.telegram.domain.TelegramBotCommand
import com.telegram.codex.shared.config.AppProperties
import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.CommandLineRunner
import org.springframework.boot.ExitCodeGenerator
import org.springframework.context.ConfigurableApplicationContext
import org.springframework.stereotype.Component

@Component
class CliTaskRunner(
    private val applicationContext: ConfigurableApplicationContext,
    private val properties: AppProperties,
    private val telegramClient: TelegramGateway,
    @param:Value("\${app.task:}") private val task: String?,
) : CommandLineRunner {
    override fun run(vararg args: String) {
        if (task.isNullOrBlank()) {
            return
        }
        if (task == "telegram:set-webhook") {
            telegramClient.setWebhook(properties.baseUrl + "/telegram/webhook", properties.telegramWebhookSecret)
        } else if (task == "telegram:update-commands") {
            telegramClient.setMyCommands(
                listOf(
                    TelegramBotCommand("status", "Bot 狀態"),
                    TelegramBotCommand("session", "目前 session 狀態"),
                    TelegramBotCommand("memory", "長期記憶狀態"),
                    TelegramBotCommand("forget", "清除長期記憶"),
                    TelegramBotCommand("compact", "壓縮目前對話 context"),
                    TelegramBotCommand("new", "新 session"),
                    TelegramBotCommand("help", "使用說明"),
                ),
            )
        } else {
            throw IllegalArgumentException("Unknown task: $task")
        }
        kotlin.system.exitProcess(org.springframework.boot.SpringApplication.exit(applicationContext, ExitCodeGenerator { 0 }))
    }
}
